### DISCUSSION TITLE
How to assign a default value for VSlider created via v_for

### DISCUSSION THREAD

**Discussion by kdasVF** [Kitware/trame#512] [Q&A] [answered]
I want to create a list of VSlide and set the default or initial value to 1 which is the max value of the slider. I tried as you see below, but the initial or default value gets set to 0. 
What am I doing wrong? 

```
vuetify.VSlider(
                                # key="i",
                                v_show="test",
                                v_for="item, i in parts",
                                label=("item.name_stem", 1),
                                v_model=(
                                    "volume_fractions[item.name_stem]",
                                    1.0,
                                ),
                                change="flushState('volume_fractions')",
                                min=0,
                                max=1,
                                step=0.1,
                                dense=True,
                                hide_details=True,
                                classes="pt-2 mx-2",
                                outlined=True,
                                ticks="always",
                                thumb_label="always",
                            )
```

**Comment by jourdain** [comment]
You can not set a default value to an expression. Technically what you are doing is creating a variable named "item.name_stem" that you initialize to 1.
item.name_stem should have been initialized in parts. And the same is true with `volume_fractions[...]` (it should have been done in volume_fractions definition).

**Reply by jourdain** [reply, marked as answer]
In other words, that initialization should be done on the server when the number of slider have changed or something...

**Reply by kdasVF** [reply]
Thank you. I am now setting the default value in volume_faction[...] when the number of sliders that is the length of volume_farction is changing, and also setting state.dirty("volume_faction") when I set the default value. This works.
